internal/backends/testing/README.md

```diff
@@ -6,3 +6,131 @@ You should use the `slint` crate instead.
+the framebuffer.
+
+For automated testing in CI environments without a windowing system / display, it might still be
+desirable to run tests. The Slint Testing Backend simulates a windowing system without requiring one:
+No pixels are rendered and text is measured by a fixed font size.
+
+Use [`init_integration_test()`] for [integration tests](https://doc.rust-lang.org/rust-by-example/testing/integration_testing.html)
+where your test code requires Slint to provide an event loop, for example when spawning threads
+and calling `slint::invoke_from_event_loop()`.
+
+Use [`init_no_event_loop()`] for [unit tests](https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html) when your test
+code does not require an event loop.
+
+## Preliminary User Interface Testing API
+
+We're developing APIs to faciliate the creation of automated tests for Slint based UIs. A building block
+is the ability to introspect and modify the state from within what would be a regular application process.
+
+A regular application might have a `main()` entry-point like so:
+
+```rust,no_run
+# slint::slint!{ export component App {} }
+fn main() -> Result<(), slint::PlatformError>
+{
+    let app = App::new()?;
+    // ... set up state, callbacks, models, ...
+
+    app.run()
+}
+```
+
+In addition, it may be desirable to create an integration test that verifies how the application behaves when simulating user input.
+The objective of the user interface testing API provided in this crate is to faciliate locating, mutation, and verifying state of
+elements in your UI. [`ElementHandle`] provides a view for these elements.
+
+The example below assumes that somehwere in the UI you have declared a `Button` with the text "Submit" and you may want to verify
+how the application behaves when simulation the activation. This is done by locating and triggering it via its accessibility interface,
+that every `Button` implements.
+
+```slint,no-preview
+import { Button } from "std-widgets.slint";
+component Form {
+    callback submit();
+    VerticalLayout {
+        // ...
+        Button {
+            text: "Submit";
+            clicked => { root.submit(); }
+        }
+    }
+}
+
+export component App {
+    callback submit <=> form.submit;
+    // ...
+    form := Form {
+        // ...
+    }
+}
+```
+
+```rust
+# use i_slint_core_macros::identity as test;
+# use i_slint_backend_testing::ElementHandle;
+# slint::slint!{
+#     import { Button } from "std-widgets.slint";
+#     component Form {
+#     callback submit();
+#     VerticalLayout {
+#         // ...
+#         Button {
+#             text: "Submit";
+#             clicked => { root.submit(); }
+#         }
+#     }
+# }
+# 
+# export component App {
+#     callback submit <=> form.submit;
+#     // ...
+#     form := Form {
+#         // ...
+#     }
+# }
+# }
+#[test]
+fn test_basic_user_interface()
+{
+    # i_slint_backend_testing::init_no_event_loop();
```

Comment:
don't we want to write slint::testing::init_no_event_loop() there?

Reply:
Good question. From a cross-platform POV you're right, that should probably be the default. I'll adjust.